src/modules/notification/infrastructure/store/notification.store.ts

```diff
@@ -1,22 +1,23 @@
 import EventNotification from '../../domain/entity/EventNotification';
-import valtioEventNotificationStore, { EventNotificationState, subscribe } from './valtio/notification.store.valtio';
+import { store } from './modules/notification';
+import { ActionTypes } from './types';
+import { MutationPayload } from 'vuex';
 
-export type StoreListener = (state: EventNotificationState) => void;
+export type StoreListener = (mutation: MutationPayload) => void;
 
 class EventNotificationStore {
   subscribeToEventNotifications(listener: StoreListener) {
-    subscribe(valtioEventNotificationStore, () => {
-      console.log(valtioEventNotificationStore);
-      listener(valtioEventNotificationStore);
+    store.subscribe((mutation: MutationPayload) => {
+      listener(mutation);
```

Comment:
You should do the mutation checks here, not in the hook. The hook should know nothing about the implementation details such as mutations.